app/scripts/lib/derived-keys.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Likewise, this file name is too generic for my taste.

I'd prefer to talk about and name the keys about what they "are" (i.e., relier keys), rather than how they're implemented (i.e., "derived keys"). 


Talking about these as "relier keys" may make more sense